test/config/Dockerfiles/contra-env-setup-test-f28/test_contra_env_setup.py

```diff
@@ -201,8 +189,13 @@ def test_jenkins_running(run_info):
             if 'jenkins' in line:
                 route = line.split()[1]
                 requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
-                request = requests.get('https://%s/login' % route, verify=False)
-                jenkins_running = True if request.status_code == 200 else False
+                for i in range(12):
```

Comment:
what's the reason for 12? is it for delay/wait?

Reply:
Yes, that's in conjuction with the 10 seconds of sleep at line 198. As the Jenkins service sometimes takes a little longer to start up after the container is initialized, I've found in testing that 120 seconds is a reasonable time to wait for it to be available.